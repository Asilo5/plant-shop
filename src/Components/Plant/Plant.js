import React from 'react';
import './Plant.scss';
import { Link } from 'react-router-dom';
import { FaSeedling } from 'react-icons/fa';

const Plant = ({_id, name, price, image}) => {

    return (
      <section className='plant-info'>
          <Link to={(`/plant/${_id}`)}>
            <img src={image} alt={name} />
            <div className='plant-name'>
                <div>
                    <p>{name}</p>
                    <p>Â£{price}</p>
                </div>
                <button><FaSeedling/></button>
            </div>
          </Link>
      </section>
    )
}

export default Plant;