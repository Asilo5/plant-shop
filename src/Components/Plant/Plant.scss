.plant-info {
    margin: 25px;
    border: 1px solid #e6d3d3;
    border-radius: 2px;
    img {
        width: 87%;
        margin: auto;
        display: block;
        margin-top: 20px;
    }
    p {
        font-size: 30px;
        margin-left: 35px;
    }
}