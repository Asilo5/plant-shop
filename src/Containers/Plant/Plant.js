import React, {Component} from 'react';
import './Plant.scss';
import { Link } from 'react-router-dom';
import { FaSeedling } from 'react-icons/fa';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { addFavouritePlant, deleteFavouritePlant } from '../../actions';
 
class Plant extends Component {

 toggleFavourite = (_id) => {
   const { favouritePlants, addFavouritePlant, deleteFavouritePlant } = this.props;
   !favouritePlants.includes((_id)) ? addFavouritePlant(_id) : deleteFavouritePlant(_id);
 }

 render () {
  const {_id, name, price, image, favouritePlants } = this.props;
  const toggleStyle = favouritePlants.includes(_id) ? 'loved-plant' : 'unloved-plant';
  console.log(favouritePlants);
  return (
    <section className='plant-info'>
        <Link to={(`/plant/${_id}`)}>
          <img src={image} alt={name} />
          <div className='plant-name'>
              <div>
                  <p>{name}</p>
                  <p>Â£{price}</p>
              </div>
              <button onClick={() => this.toggleFavourite(_id)} className={toggleStyle} ><FaSeedling/></button>
          </div>
        </Link>
    </section>
  )

 }
}

export const mapStateToProps = ({ favouritePlants }) => ({
  favouritePlants
});

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({ 
    addFavouritePlant,
    deleteFavouritePlant
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(Plant);