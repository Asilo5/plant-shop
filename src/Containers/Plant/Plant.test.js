import React from 'react';
import { shallow } from 'enzyme';
import {Plant} from './Plant';

describe('Plant component', () => {
    it('should match snapshot', () => {
        let wrapper = shallow(<Plant _id={1} 
                                     name='Aloe'
                                     price={15} 
                                     image='image link' />)
        expect(wrapper).toMatchSnapshot();
    })
})