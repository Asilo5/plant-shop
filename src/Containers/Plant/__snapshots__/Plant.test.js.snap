// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plant component should match snapshot 1`] = `
<section
  className="plant-info"
>
  <Link
    to="/plant/1"
  >
    <img
      alt="Aloe"
      src="image link"
    />
    <div
      className="plant-name"
    >
      <div>
        <p>
          Aloe
        </p>
        <p>
          Â£
          15
        </p>
      </div>
      <button>
        <FaSeedling />
      </button>
    </div>
  </Link>
</section>
`;
